File: bbclient/src/com/bbclient/SetRunnable.java
package com.bbclient;

import com.hazelcast.map.IMap;

class SetRunnable extends IMapMethodRunnable {
    private final String value;

    /*
     * synchronized
     */
    private int nextKey;
    private int lastKey;

    private synchronized int getNextKey() {
        lastKey = nextKey;
        nextKey++;
        return lastKey;
    }

    synchronized int getLastKey() {
        return lastKey;
    }

    SetRunnable(IMap<Integer, String> map,
                int mapValueSize,
                int firstKey) {
        super(map, "set", true);
        this.value = "*".repeat(mapValueSize);
        this.nextKey = firstKey;
        this.lastKey = this.nextKey;
    }

    void invokeMethod() {
        map.set(getNextKey(), value);
    }
}

File: bbclient/src/com/bbclient/SetRunnable.kt
package com.bbclient

import com.hazelcast.map.IMap

internal class SetRunnable(
    map: IMap<Int?, String?>,
    mapValueSize: Int,
    /*
     * synchronized
     */private var nextKey: Int
) : IMapMethodRunnable(map, "set", true) {
    private val value = "*".repeat(mapValueSize)

    @get:Synchronized
    var lastKey: Int
        private set

    @Synchronized
    private fun getNextKey(): Int {
        lastKey = nextKey
        nextKey++
        return lastKey
    }

    init {
        this.lastKey = this.nextKey
    }

    override fun invokeMethod() {
        map[getNextKey()] = value
    }
}
